### PR TITLE
refactor(core): update cloud dependency, cloud client calls

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

update cloud dependency, cloud client calls
1. we updated the API interface in [previous cloud changes](https://github.com/logto-io/cloud/pull/617/files#diff-ee801056b7210a92bd716efeaeff4468c9eda32a3886983df032b85a48075f83), as a result, we need to update the API calls in Logto core
2. refactor the way to get custom JWT deploy script object

## Testing

Tested locally and add complete unit test for the refactor custom JWT deploy script util method.

## Checklist

- [x] unit tests
~~- [ ] `.changeset`~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/core/src/libraries/logto-config.ts [hunk 1]
```diff
@@ -160,17 +164,22 @@ export const createLogtoConfigLibrary = ({
...
+       * Only add `/test` endpoint for Cloudflare workers when testing.
+       * O/w overwrite the existing JWT customizer script.
+       */
+      [payload.key]: payload.isTest
+        ? {
+            test: payload.value.script,
+          }
+        : {
+            production: payload.value.script,
+          },
```
Comment (gao-sun) [inline packages/core/src/libraries/logto-config.ts:178]: it looks like the comment doesn't explain the code

File: packages/core/src/libraries/logto-config.ts [hunk 2]
```diff
@@ -160,17 +164,22 @@ export const createLogtoConfigLibrary = ({
...
     const customizerScriptsFromDatabase = getJwtCustomizerScripts(jwtCustomizers);
 
-    const newCustomizerScripts: { [key in LogtoJwtTokenKey]?: string } = {
-      [payload.key]: payload.value.script,
+    const newCustomizerScripts: CustomJwtDeployRequestBody = {
+      /**
+       * Only add `/test` endpoint for Cloudflare workers when testing.
+       * O/w overwrite the existing JWT customizer script.
+       */
+      [payload.key]: payload.isTest
```
Comment (gao-sun) [inline packages/core/src/libraries/logto-config.ts:172]: non-blocking: i doubt if `isTest` is good - how about we just pass a `useCase: 'test' | 'production'` and use it directly?

File: packages/core/src/libraries/logto-config.script-merging.test.ts [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                },
+              }
+        );
+      }
+    );
+
+    it.each(Object.values(LogtoJwtTokenKey))(
+      `when ${LogtoJwtTokenKey.ClientCredentials} script is working, deploy $s script`,
+      (deployingKey) => {
+        const workingKey = LogtoJwtTokenKey.ClientCredentials;
```
Comment (gao-sun) [inline packages/core/src/libraries/logto-config.script-merging.test.ts:199]:
```suggestion
        const existingKey = LogtoJwtTokenKey.ClientCredentials;
```

File: packages/core/src/libraries/logto-config.script-merging.test.ts [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+      });
+    });
+  });
+
+  describe('Deploy script:', () => {
+    // Test it.each() can not be nested, so we have to test each key separately.
+    it.each(Object.values(LogtoJwtTokenKey))(
+      `when ${LogtoJwtTokenKey.AccessToken} script is working, deploy $s script`,
+      (deployingKey) => {
+        const workingKey = LogtoJwtTokenKey.AccessToken;
```
Comment (gao-sun) [inline packages/core/src/libraries/logto-config.script-merging.test.ts:165]:
```suggestion
        const existingKey = LogtoJwtTokenKey.AccessToken;
```

File: packages/core/src/libraries/logto-config.ts (hunk 2 above)
Comment (darcyYe) [reply packages/core/src/libraries/logto-config.ts]: Either is fine for me, but seems `useCase` could be more  straightforward. Updated.

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/libraries/logto-config.ts [hunk 5]
```diff
@@ -160,17 +164,31 @@ export const createLogtoConfigLibrary = ({
...
+       * previous handler of the API endpoint.
+       */
+      [payload.key]:
+        payload.useCase === 'test'
+          ? {
+              test: payload.value.script,
+            }
+          : {
+              production: payload.value.script,
+            },
```
[the remark this answers is not in the document]
Comment (darcyYe) [reply packages/core/src/libraries/logto-config.ts]: Updated

**Pull request merged**